hassio/src/addon-view/config/hassio-addon-config.ts

```diff
@@ -71,14 +112,37 @@ class HassioAddonConfig extends LitElement {
 
   protected updated(changedProperties: PropertyValues): void {
     super.updated(changedProperties);
-    if (changedProperties.has("addon")) {
-      this._editor.setValue(this.addon.options);
+    if (changedProperties.has("addon") || changedProperties.has("_yamlMode")) {
+      this._canShowSchema = !this.addon.schema.find(
+        (entry) => !SUPPORTED_UI_TYPES.includes(entry.type)
+      );
+
+      const editor = this._editor;
+      if (editor) {
+        editor.setValue(this.addon.options);
+      }
+    }
+  }
+
+  private _handleAction(ev: CustomEvent<ActionDetail>) {
+    switch (ev.detail.index) {
+      case 0:
+        this._yamlMode = !this._yamlMode;
+        break;
     }
   }
 
   private _configChanged(ev): void {
-    this._configHasChanged = true;
-    this._valid = ev.detail.isValid;
+    if (this.addon.schema && this._canShowSchema) {
+      this._valid = true;
+      this._configHasChanged = true;
+    } else {
+      this._configHasChanged = true;
+      this._valid = ev.detail.isValid;
+    }
+    if (this._valid && this._configHasChanged) {
```

Comment:
`this._configHasChanged` will always be true here

Reply:
That is `true`